packages/custom_lint_core/lib/src/configs.dart

```diff
@@ -35,11 +37,19 @@ class CustomLintConfigs {
     final include = yaml['include'] as Object?;
     var includedOptions = CustomLintConfigs.empty;
     if (include is String) {
-      final includeAbsolutePath = absolute(
-        analysisOptionsFile.parent.path,
-        include,
-      );
-      includedOptions = CustomLintConfigs.parse(
+      var includeUri = Uri.parse(include);
+      includeUri = (await Isolate.resolvePackageUri(includeUri)) ?? includeUri;
+
+      final includeAbsolutePath = includeUri.isAbsolute
+          ? includeUri.toFilePath()
+          : normalize(
+              absolute(
+                analysisOptionsFile.parent.path,
+                includeUri.path,
+              ),
+            );
```

Comment:
What is this about? I don't understand the goal of those changes.

Reply:
We shouldn't have to use resolvePackageUri.
We should be able to manually match that `uri` against a package resolved in the package_config.json

Reply:
Thanks a lot for your suggestion! After checking out the [documentation](https://api.flutter.dev/flutter/dart-isolate/Isolate/resolvePackageUri.html) for the resolvePackageUri method, it seems to align perfectly with your description, as it examines the package URI in the project's package_config.json. 
If I've misunderstood your point, could you please clarify?

Reply:
Also, i've refactored this code a little so now it looks more understandable
.

Reply:
Custom_lint already decodes package_config.jsons separately. Using resolvePackageUri is redundant in that sense.

Also, using resolvePackageUri forces the constructor to be asynchronous, when it wasn't before. I'd prefer not using it and keep the constructor synchronous

Reply:
I've gotten rid of the `resolvePackageUri` method in the last commit and used the project's `PackageUtils` so now the solution seems to fit with your suggestion. 
Looking forward to your feedback!

Reply:
Hey @rrousselGit, could you please take a look at the last changes? Would be very appreciated

Reply:
Hi @rrousselGit, happy holidays! Any chance you can take a look at it one more time soon?